src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+import java.util.ArrayList;
+
+/**
+ *  A class which encapsulates the task list and operations on the list.
+ *  @author  Chen Guanzhou
+ *  @version v1
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Removes the specified task from the task list.
+     * @param remove The task to be removed.
+     */
+    public void removeTask(Task remove) {
+        this.taskList.remove(remove);
+    }
+
+    /**
+     * Adds a task to the end of the task list.
+     * @param t The task to be added.
+     */
+    public void addTask(Task t) {
+        this.taskList.add(t);
+    }
+
+    
+    public TaskList matchingItems(String input) {
```

Comment:
Should the method name be getMatchingItems()? Besides, perhaps add a header comment to this method?

Reply:
Good points!